backend/icpy/agent/tools/atlascloud/models.py

```
@@ -0,0 +1,223 @@
+        priority_keys = ['download', 'video_url', 'file', 'url', 'output', 'video']
+        for key in priority_keys:
+            if key in data:
+                value = data[key]
+                if isinstance(value, str) and self._is_direct_file_url(value):
+                    return value
+                elif isinstance(value, list) and value:
+                    for item in value:
+                        if isinstance(item, str) and self._is_direct_file_url(item):
+                            return item
+        return None
+        return any(indicator in url for indicator in api_indicators)
```

Ruff (0.14.14)

[error] Undefined name `url`

(F821)